pysam/libcalignmentfile.pyx

```diff
@@ -1348,7 +1541,7 @@ cdef class AlignmentFile(HTSFile):
 
```

Comment:
Add tid translation here for when `read.header is not self.header`?  And add a `translate` method to `AlignedSegment`.

This is a breaking change that needs to happen at some point, because it breaks apps that have already translated their tids.